plugins/modules/ndo_schema_template_contract_service_chain.py

```diff
@@ -0,0 +1,381 @@
+                device=dict(
+                    type="dict",
+                    options=dict(
+                        name=dict(type="str", required=True),
+                        template=dict(type="str"),
+                        template_id=dict(type="str"),
+                    ),
+                    mutually_exclusive=[["template", "template_id"]],
+                    required_one_of=[["template", "template_id"]],
+                ),
+                provider_interface_name=dict(type="str", required=True, aliases=["provider_interface", "provider"]),
+                provider_redirect=dict(type="bool"),
+                consumer_interface_name=dict(type="str", required=True, aliases=["consumer_interface", "consumer"]),
+                consumer_redirect=dict(type="bool"),
+            ),
+            mutually_exclusive=[["uuid", "device"]],
+            required_one_of=[["uuid", "device"]],
+        ),
+        state=dict(type="str", default="present", choices=["absent", "present", "query"]),
+    )
+
+    module = AnsibleModule(
+        argument_spec=argument_spec,
+        supports_check_mode=True,
+        required_if=[
+            ["state", "present", ["service_nodes"]],
+        ],
+        mutually_exclusive=[["schema", "schema_id"], ["template", "template_id"], ["contract", "contract_uuid"]],
+        required_one_of=[["contract", "contract_uuid"], ["template", "template_id"]],
+    )
+
+    schema = module.params.get("schema")
+    schema_id = module.params.get("schema_id")
+    template = module.params.get("template")
+    template_id = module.params.get("template_id")
+    contract = module.params.get("contract")
+    contract_uuid = module.params.get("contract_uuid")
+    node_filter = CONTRACT_SERVICE_CHAIN_NODE_FILTER_MAP.get(module.params.get("node_filter"))
+    service_nodes = module.params.get("service_nodes")
+    state = module.params.get("state")
+
+    mso = MSOModule(module)
+
+    if not template_id and not (schema or schema_id):
+        mso.fail_json(msg="The schema or schema_id is required when the template is set.")
+
+    mso_templates = MSOTemplates(mso)
+    mso_template = MSOTemplate(mso, "application", template, template_id, schema_name=schema, schema_id=schema_id)
+
+    template_id = template_id or mso_template.template.get("templateId")
+    template = template or mso_template.template.get("appTemplate", {}).get("template", {}).get("name")
+    contract_match = mso_template.get_application_template_contract(contract_uuid, contract, fail_module=True)
+    service_chain = contract_match.details.get("serviceChaining") if contract_match and contract_match.details.get("serviceChaining") else {}
+
+    reference_collections = {
+        "serviceDeviceRef": {
+            "name": "deviceName",
+            "reference": "deviceRef",
+            "type": "serviceDevice",
+            "template": "deviceTemplateName",
+            "templateId": "deviceTemplateId",
+        },
+    }
```

Comment:
add blank line between this line and the if

Reply:
Done!